bottlecap/src/lifecycle/invocation_times.rs

```diff
@@ -20,11 +23,11 @@ impl InvocationTimes {
         self.head = (self.head + 1) % LOOKBACK_COUNT;
     }
 
-    pub(crate) fn should_adapt_to_periodic(&self, now: u64) -> bool {
+    pub(crate) fn should_adapt(&self, now: u64, flush_timeout: u64) -> FlushStrategy {
```

Comment:
Maybe we rename this to something more significant now that we are returning the strategy?